server/schemas/badge.schema.js

```diff
@@ -0,0 +1,19 @@
+import Joi from '@hapi/joi';
+import { requiredObjectId, requiredString, optionalString } from './helper.schema';
+import { BADGE_ACCESS_LEVEL } from '../helpers/constants';
+
+const userLevel = Joi.number()
+  .label('Step')
```

Comment:
Using `Step` here looks awkward and difficult to understand. Let's update the `userLevel` to `assignedRole` to make it more relatable.

```suggestion
  .label('Role')
```